### DISCUSSION TITLE
Allow "raw" typescript as custom endpoint source

### DISCUSSION THREAD

**Discussion by LittleBigBug** [directus/directus#8324] [Feature Requests]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

I couldn't find another issue that describes this.
You can't make extensions, or endpoints, using typescript

### To Reproduce

Create a new Directus API project, and in `extensions/endpoints/` create a folder and name it anything, and create a file in it named `index.ts`

Use `yarn directus start` to start directus and notice:

```
14:52:01 ✨ Loaded extensions: terminals
14:52:01 ⚠️  Database migrations have not all been run
14:52:01 ✨ Loaded extensions: terminals
14:52:01 ⚠️  Couldn't register endpoint "terminals"
14:52:01 ⚠️  Cannot find module '/home/ethan/Projects/goairheads-api/extensions/endpoints/terminals/index.js'
Require stack:
- /home/ethan/Projects/goairheads-api/node_modules/directus/dist/extensions.js
```

### What version of Directus are you using?

44a5ca208aca078b38e06a35e21e85c79ad9fa33

### What version of Node.js are you using?

v16.10.0

### What database are you using?

MariaDB

### What browser are you using?

Firefox

### What operating system are you using?

Windows with Ubuntu 20 (WSL)

### How are you deploying Directus?

Local dev

**Comment by rijkvanzanten** [comment]
Directus uses Node's require system, so valid JavaScript is required

**Reply by LittleBigBug** [reply]
Yes, would using something like `ts-node` be possible to transpile at runtime like this?
I realize now that I could compile it on the extension's end as well but I still figure it's a useful suggestion considering it'd make using Typescript a lot easier and more feasible to new Directus users who want to extend it.

Also, thanks for moving it to discussions. I wasn't familiar with how it was to be organized.

**Reply by nickrum** [reply]
Using `ts-node` wouldn't help unfortunately, as we have to require/import the extension source directly. This means we would have to transpile the typescript source to javascript and write it to disc before it can be loaded. IMO it's easier to leave this as a build step during extension development.

The extensions-sdk already supports scaffolding and building typescript extensions, by the way.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
